docs/src/pages/demos/tables/ReactVirtualizedTable.tsx

```diff
@@ -0,0 +1,228 @@
+      [SortDirection.DESC]: 'desc',
+    };
+
+    const inner =
+      !columns[columnIndex].disableSort && sort != null ? (
+        <TableSortLabel active={dataKey === sortBy} direction={direction[sortDirection]}>
+          {label}
+        </TableSortLabel>
+      ) : (
+        label
+      );
+
+    return (
+      <TableCell
+        component="div"
+        className={clsx(classes.tableCell, classes.flexContainer, classes.noClick)}
+        variant="head"
+        style={{ height: headerHeight }}
+        align={columns[columnIndex].numeric || false ? 'right' : 'left'}
+      >
+        {inner}
+      </TableCell>
+    );
+  };
+
+  render() {
+    const { classes, columns, ...tableProps } = this.props;
+    return (
+      <AutoSizer>
+        {({ height, width }) => (
+          <Table height={height} width={width} {...tableProps} rowClassName={this.getRowClassName}>
+            {columns.map(({ cellContentRenderer = null, className, dataKey, ...other }, index) => {
+              let renderer;
+              if (cellContentRenderer != null) {
+                renderer = cellRendererProps =>
+                  this.cellRenderer({
+                    cellData: cellContentRenderer(cellRendererProps),
+                    columnIndex: index,
+                  });
+              } else {
+                renderer = this.cellRenderer;
+              }
+
+              return (
+                <Column
+                  key={dataKey}
+                  headerRenderer={headerProps =>
+                    this.headerRenderer({
+                      ...headerProps,
+                      columnIndex: index,
+                    })
+                  }
+                  className={clsx(classes.flexContainer, className)}
+                  cellRenderer={renderer}
+                  dataKey={dataKey}
+                  {...other}
+                />
+              );
+            })}
+          </Table>
+        )}
+      </AutoSizer>
+    );
+  }
+}
+
+const VirtualizedTable = withStyles(styles)(MuiVirtualizedTable);
+
+// ---
+
+const data: Array<[string, number, number, number, number]> = [
+  ['Frozen yoghurt', 159, 6.0, 24, 4.0],
+  ['Ice cream sandwich', 237, 9.0, 37, 4.3],
+  ['Eclair', 262, 16.0, 24, 6.0],
+  ['Cupcake', 305, 3.7, 67, 4.3],
+  ['Gingerbread', 356, 16.0, 49, 3.9],
+];
+
+interface DataRow {
+  id: number;
+  dessert: string;
+  calories: number;
+  fat: number;
+  carbs: number;
+  protein: number;
+}
+
+function createData(id: number, dessert: string, calories: number, fat: number, carbs: number, protein: number) {
+  return { id, dessert, calories, fat, carbs, protein };
+}
+
+const rows: DataRow[] = [];
+
+for (let i = 0; i < 200; i += 1) {
+  const randomSelection = data[Math.floor(Math.random() * data.length)];
+  rows.push(createData(i, ...randomSelection));
+}
+
+function ReactVirtualizedTable() {
+  return (
+    <Paper style={{ height: 400, width: '100%' }}>
+      <VirtualizedTable
```

Comment:
I also don't understand how to migrate this error correctly.

Type '{ rowCount: number; rowGetter: ({ index }: { index: number; }) => DataRow; onRowClick: (event: Event) => void; columns: ({ width: number; flexGrow: number; label: string; dataKey: string; } | { width: number; label: string; dataKey: string; numeric: boolean; })[]; }' is not assignable to type 'IntrinsicAttributes & Pick<{}, never> & StyledComponentProps<"flexContainer" | "tableRow" | "tableRowHover" | "tableCell" | "noClick"> & { children?: ReactNode; }'.